load-test-framework/src/main/java/com/google/pubsub/flic/Driver.java

```diff
@@ -220,18 +273,78 @@ private void run() {
           log.info("===============================================");
         }
       }, 5, 5, TimeUnit.SECONDS);
-      // Wait for the load test to finish.
-      gceController.waitForClients();
+      Map<ClientType, Controller.LoadtestStats> statsMap;
+      AtomicDouble publishLatency = new AtomicDouble(0);
+      // This should be a number_of_messages test so that we know when the publisher is done.
```

Comment:
It could be either, this was a comment I wrote while I was thinking things through, meant to delete it. Done.